File: backend/src/main/java/com/project/entities/DiaryEntry.java
package com.project.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "diary_entries")
@Getter
@Setter
public class DiaryEntry extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "diary_id")
    private Long diaryId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "encrypted_text", columnDefinition = "TEXT", nullable = false)
    private String encryptedText;

    @Column(name = "is_deleted")
    private boolean deleted = false;

   

}

File: backend_dotnet/SafeMind.Core/Entities/DiaryEntry.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace SafeMind.Core.Entities
{
    [Table("diary_entries")]
    public class DiaryEntry : BaseEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("diary_id")]
        [JsonPropertyName("diaryId")]
        public long DiaryId { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        [Column("user_id")]
        [JsonPropertyName("userId")]
        public long UserId { get; set; }

        [JsonPropertyName("user")]
        public User User { get; set; } = null!;

        [Required]
        [Column("encrypted_text", TypeName = "TEXT")]
        [JsonPropertyName("encryptedText")]
        public string EncryptedText { get; set; } = string.Empty;

        [Column("is_deleted")]
        [JsonPropertyName("isDeleted")]
        public bool IsDeleted { get; set; } = false;
    }
}
